File: Karan/BackUpBackEnd/HomeAppliancesSpring/src/main/java/com/example/demo/entities/Category.java
package com.example.demo.entities;

import java.util.HashSet;
import java.util.Set;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name="category")
public class Category {
	
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	int cat_id;
	
	@Column
	String type;
	
	@Column
	String description;
	
	@OneToMany(mappedBy="category",cascade=CascadeType.ALL)
	@JsonIgnoreProperties("category")
	Set<Appliance> appliances=new HashSet<>();
	
	
	

}

File: Main Project/Backend .net/RentIT/Models/Category.cs
using System;
using System.Collections.Generic;

namespace RentIT.Models
{
    public partial class Category
    {
        public Category()
        {
            Appliances = new HashSet<Appliance>();
        }

        public int CatId { get; set; }
        public string? Description { get; set; }
        public string? Type { get; set; }

        public virtual ICollection<Appliance> Appliances { get; set; }
    }
}
